components/scream/src/share/field/field_manager.cpp

```diff
@@ -53,11 +36,15 @@ void FieldManager::register_field (const FieldRequest& req)
     m_fields[id.name()] = std::make_shared<Field>(id);
   } else {
     // Make sure the input field has the same layout and units as the field already stored.
+    // NOTE: we allow the input units to be "invalid", in case the customer requesting
```

Comment:
In the usage of the new diagnostic isn't it possible to add the field units to the parameter list that is passed at construction, so that we can retain the units?  Presumably, if someone is only looking a output file with surface values they won't want to have to query the default output to see what the units should have been.

Reply:
I can add that.